xwiki-platform-core/xwiki-platform-livedata/xwiki-platform-livedata-webjar/src/main/webjar/mockData.js

```diff
@@ -54,30 +58,46 @@ define({
       {
         id: "doc_title",
         name: "Name",
-        type: "Text",
+        type: "text",
```

Comment:
I don't think you can merge property type and property filter. They are two different things. Property type is normally mapped to an xclass property type, while filter can be custom. Moreover we can have multiple filters that can work with a property type (date range and text fitlers can both work with a date property) or a filter that can work with multiple property types (text filter can work with both String and Date properties).

Reply:
I think the fallback should be: property descriptor -> property type -> default displayer / filter. This means:

* we provide a default displayer / filter
* a property type can specify a displayer / filter with additional configuration options
* a property descriptor can specify a displayer / filter with additional configuration options

So for instance the Date property type could use the date-range filter, while for a specific live data instance, we may want to use a different filter for a specific date property or use the date-range filter with some custom configuration.

The JSON configuration produced by the live data macro will include:
* the displayer / filter in the property descriptor, if the developer has explicitly specified this
* the displayer / filter in the known property types (this is done automatically, the developer that calls the macro doesn't care about this)